### PR TITLE
Implement UDP command.

### PR DESCRIPTION
**Pull request by yuguorui**
Hi, I have been using fast-socks5 for quite a while, and it works great. 

It's just a pity that UDP support is missing, I've done some work here, could you please write some comments?

### CODE REVIEW

**Comment by dizda** [comment]
Hi @yuguorui,

Thank you for your PR. 
Give me some time to review it, then I'll get back to you.

File: src/client.rs [hunk 1]
```diff
@@ -236,69 +247,76 @@ where
...
+
+                    packet[3] = 0x04;
+                    debug!("addr ip {:?}", (*addr.ip()).octets());
+                    packet[4..20].copy_from_slice(&(addr.ip()).octets()); // ip
+                    packet[20..padding].copy_from_slice(&addr.port().to_be_bytes());
+                    // port
+                }
+                TargetAddr::Domain(ref domain, port) => {
+                    debug!("TargetAddr::Domain");
+                    if domain.len() > u8::max_value() as usize {
```
Comment (dizda) [inline src/client.rs:308]: `u8::max_value()` is deprecated, could you replace it with MAX?

**Comment by dizda** [comment]
It looks really good, thank you!
I don't have any real case scenario to test the UDP way though.

**Comment by dizda** [comment]
Could you also update the README.md?

**Comment by dizda** [comment]
Would it be possible to add a flag on the server::Config to toggle UDPAssociate support?

**Comment by dizda** [comment]
I've merged that PR, you can submit new changed onto a new one 🙏

Thank you.

File: src/client.rs (hunk 1 above)
Comment (yuguorui) [reply src/client.rs:308]: Sure, I will fix that.

**Comment by yuguorui** [comment]
> Could you also update the README.md?

Of course, it's my pleasure.

**Comment by yuguorui** [comment]
> Would it be possible to add a flag on the server::Config to toggle UDPAssociate support? (in order to disallow it)

Yes, I also think it's a good idea to avoid code bloat, I'll do it asap.

**Comment by dizda** [comment]
Thank you @yuguorui

**Pull request merged**